src/redprl/syntax.sml

```diff
@@ -111,7 +111,9 @@ struct
               | (O.TAG_DPROD, (A :: xB :: args)) => (O.MONO O.DPROD $$ [A, xB], args)
               | (O.TAG_PATH, (uA :: a0 :: a1 :: args)) => (O.MONO O.PATH_TY $$ [uA, a0, a1], args)
               | _ => raise Fail "Syntax.out hcom: Malformed tag"
-           val (_ \ cap) :: args = args
+           val (cap , args) = (case args of
+                                    ((_ \ cap) :: args) => (cap, args)
+                                  | _ => raise Fail "Syntax.out hcom: Missing cap")
```

Comment:
I would leave the partial binding as-is, rather than adding this case statement, since it is an invariant that this will never happen; I prefer the simple code. If you look at the refiner / rules, there are hundreds of partial matches which are essential for the rules being readable, for instance.

Also, I prefer the following indentation style:

```sml
val (cap , args) = 
  (case args of
      ((_ \ cap) :: args) => (cap, args)
    | _ => raise Fail "Syntax.out hcom: Missing cap")
 ```

The main idea of the indentation style that I prefer is that the indentation of some code should never depend on the *lexical* width of some other identifier; this means that I don't try and "line things up", but rather use indentation as a way to signify syntactic hierarchy. I think that there are some inevitable exceptions, due to ML's keywords, but in general I try to follow that rule.

And a number of redundant parentheses could also be removed, leading to:

```sml
val (cap , args) = 
  case args of
     (_ \ cap) :: args => (cap, args)
   | _ => raise Fail "Syntax.out hcom: Missing cap"
 ```